docs/02-guides/translate-your-application.mdx

```diff
@@ -0,0 +1,303 @@
+
+:::info
+
+You can find the full set of components in the
+[React Intl Components documentation](https://formatjs.io/docs/react-intl/components/).
+
+:::
+
+For instance, let's see how to transform your values in `react-intl` Components.
+
+### Strings
+
+```tsx
+import { FormattedMessage } from "react-intl";
+
+<FormattedMessage
+  id="My.Component.Id.messageId"
+  defaultMessage="Default translation"
+/>;
+```
+
+:::note
+
+To take full advantage of the `translate` cli in Front-Commerce, a default
+message is required for formatted messages. This will be used to extract the
+messages for the default locale and ensure there are no dead translations, or
+missing translations.
+
+:::
+
+### Dates
+
+```tsx
+import { FormattedDate } from "react-intl";
+
+<FormattedDate value={date} year="numeric" month="long" day="2-digit" />;
+```
+
+### Prices
+
+```tsx
+import { FormattedNumber } from "react-intl";
+
+<FormattedNumber
+  value={value.amount}
+  style="currency"
+  currency={value.currency}
+/>;
+```
+
+:::note
+
+However, for this particular use case we have abstracted this in Front-Commerce
+with the
+[theme/components/atoms/Price](https://gitlab.blackswift.cloud/front-commerce/front-commerce/-/blob/main/packages/theme-chocolatine/theme/components/atoms/Typography/Price/index.js)
+component (and its variants: `ProductPrice`, `PriceOrFree`). We recommend you to
+use it instead, so you could benefit from better integration with its related
+GraphQL Type.
+
+:::
+
+### Without React Component
+
+These components will be enough in most cases. However, the issue with these is
+that it wraps your string within a React Element, which may be troublesome when
+you actually need to handle the string itself.
+
+For instance, this is the case when you want to add some label attributes to
+your <abbr title="Document Object Model">DOM</abbr> elements.
+
+```jsx
+<span class="icon" aria-label="Icon title displayed for screen readers" />
+```
+
+Fortunately, this is correctly handled by `react-intl` if you use
+[`defineMessages`](https://formatjs.io/docs/react-intl/api/#definemessagesdefinemessage)
+combined with the
+[`useIntl`](https://formatjs.io/docs/react-intl/api/#useintl-hook) hook or the
+[`injectIntl`](https://formatjs.io/docs/react-intl/upgrade-guide-3x/#new-useintl-hook-as-an-alternative-of-injectintl-hoc)
+HOC.
+
+```mdx-code-block
+<Tabs>
+<TabItem value="useIntl">
+```
+
+You can learn more about hooks in the
+[react documentation](https://reactjs.org/docs/hooks-intro).
+
+```jsx
+import { defineMessages, useIntl } from "react-intl";
+
+const messages = defineMessages({
+  ariaLabel: {
+    id: "screen-reader-icon-title",
+    defaultMessage: "Icon title displayed for screen readers",
+  },
+});
+
+const MyComponentWithHook = (props) => {
+  const intl = useIntl();
+  return (
+    <span class="icon" aria-label={intl.formatMessage(messages.ariaLabel)} />
+  );
+};
+```
+
+```mdx-code-block
+</TabItem>
+<TabItem value="injectIntl">
```

Comment:
**thought:** Do we want to recommend HOCs? I feel that an "older, less preferred" way to do things among the community. Maybe we could just add the HOC way in a `:::info` with a link to the docs, instead of a dedicated tab?

This comment is completely non-blocking btw, I'm just wondering.